### PR TITLE
feat: enable sourcemaps generation

### PR DESCRIPTION
**Pull request by ekashida**
## Does this pull request introduce a breaking change?

No

## Does this pull request introduce an observable change?

No

## GUS work item

W-12488841

### CODE REVIEW

File: packages/@lwc/engine-dom/scripts/rollup.config.js [hunk 1]
```diff
@@ -68,12 +69,16 @@ function injectInlineRenderer() {
...
-                return source.replace(
-                    RENDERER_REPLACEMENT_STRING,
-                    code.replace('var renderer = ', '')
-                );
+                magic.replace(RENDERER_REPLACEMENT_STRING, code.replace('var renderer = ', ''));
+
+                return {
+                    code: magic.toString(),
+                    map: magic.generateMap(),
+                };
```
Comment (nolanlawson) [inline packages/@lwc/engine-dom/scripts/rollup.config.js:81]: Awesome!

File: packages/@lwc/aria-reflection/tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,6 @@
     "extends": "../../../tsconfig.json",
 
     "compilerOptions": {
-        "sourceMap": false,
```
Comment (nolanlawson) [inline packages/@lwc/aria-reflection/tsconfig.json:5]: At first I was confused by this, but it looks like removing this line from `tsconfig.json` has no effect, either with or without Rollup. I checked, and all our packages have the proper `.map` files in the `dist/`.

**Review by nolanlawson — COMMENTED** [review]
This all LGTM, except I wonder if we need to have `.map` files in the `lwc` package as well? This is commonly where people grab our files from, and I noticed there are no `.map`s there.

**Review by jye-sf — APPROVED** [review]

**Review by jye-sf — APPROVED** [review]
I see the `.map` files in the `lwc` package now.

**Comment by nolanlawson** [comment]
I was trying to test this in Karma, which seems to automatically inject the `.map` file into a base64-encoded data URL at the end of the file. (Great!) However, I noticed that the individual JS files were actually not shown in the Chrome DevTools, e.g. when I set a breakpoint inside of `synethetic-shadow.js`.

Using [source-map-explorer](https://www.npmjs.com/package/source-map-explorer), I noticed that it looks like the sourcemap only includes a single file entry (for files in the `lwc` package):

Screen Shot 2023-02-15 at 2 01 59 PM

Maybe the reason this is happening is because the sourcemaps for the `@lwc/synthetic-shadow` / `@lwc/engine-dom` packages are corrupt?

Screen Shot 2023-02-15 at 2 04 11 PM

I haven't looked into this further, but it would be great to see the individual files when debugging our Karma tests.

**Comment by ekashida** [comment]
Will address the issue pointed out by @nolanlawson in a separate PR

**Pull request merged**